core/arch/arm/mm/core_mmu.c

```diff
@@ -935,6 +935,8 @@ static bool assign_mem_va(vaddr_t tee_ram_va,
 			map->va = va;
 			if (ADD_OVERFLOW(va, map->size, &va))
 				return false;
+			if (va & ~va_mask)
+				return false;
```

Comment:
This is needed for 64bit VA space. I suggest to not add `va_mask` as argument to this function and use here:
```c
			if (IS_ENABLED(ARM64) && va >= BIT(get_va_width()))
				return false;
```
and same the the change below.